### PR TITLE
Rule/yoda

### PR DESCRIPTION
**Pull request by rowbi-wan**
## Suggested rule/changes(s):
* `<yoda>` 

## Reason for addition/amendment
Discourage code typed like yoda would speak

```
if ("red" === colour) {
    //...
}
```

@netsells/frontend - Please review

### CODE REVIEW

File: rules/javascript.js [hunk 1]
```diff
@@ -25,5 +25,9 @@ module.exports = {
         }],
         // Discourage using 'var' for creating variables - require using let/const instead
         'no-var': _THROW.ERROR,
+        // Discourage code typed like yoda would speak
+        'yoda': [_THROW.ERROR, 'never', {
+            'exceptRange': true,
```
Comment (samturrell) [inline rules/javascript.js]: no quotes

File: readme.md [hunk 2]
```diff
@@ -128,6 +128,51 @@ function add(x, y) {
...
 ---
 
+#### 📍 yoda
+Discourage code typed like yoda would speak
+
+##### ❌ Example of incorrect code for this rule:
+
+```js
+
+if ("red" === color) {
```
Comment (samturrell) [inline readme.md]: single quotes

File: readme.md [hunk 3]
```diff
@@ -128,6 +128,51 @@ function add(x, y) {
...
+
+##### ✅ Example of correct code for this rule:
+
+```js
+
+if (5 & value) {
+    // ...
+}
+
+if (value === "red") {
```
Comment (samturrell) [inline readme.md]: single quotes (should make this a linter rule)

**Comment by wik-z** [comment]
An important rule, this is.

**Review by samturrell — CHANGES_REQUESTED** [review]
I like it

**Review by samturrell — APPROVED** [review]

**Pull request merged**